eth2util/rlp/rlp_test.go

```diff
@@ -0,0 +1,102 @@
+// Copyright © 2022 Obol Labs Inc.
+//
+// This program is free software: you can redistribute it and/or modify it
+// under the terms of the GNU General Public License as published by the Free
+// Software Foundation, either version 3 of the License, or (at your option)
+// any later version.
+//
+// This program is distributed in the hope that it will be useful, but WITHOUT
+// ANY WARRANTY; without even the implied warranty of  MERCHANTABILITY or
+// FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+// more details.
+//
+// You should have received a copy of the GNU General Public License along with
+// this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package rlp_test
+
+import (
+	"fmt"
+	"math/rand"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/obolnetwork/charon/eth2util/rlp"
+	"github.com/obolnetwork/charon/testutil"
+)
+
+// TestBytes tests encoding and decoding of byte slices using examples from the RLP spec.
+// See https://ethereum.org/en/developers/docs/data-structures-and-encoding/rlp/#examples.
+func TestBytes(t *testing.T) {
+	tests := []struct {
+		input  []byte
+		output []byte
+	}{
+		{
+			input:  []byte("dog"),
+			output: []byte{0x83, byte('d'), byte('o'), byte('g')},
+		},
+		{
+			input:  []byte(""),
+			output: []byte{0x80},
+		},
+		{
+			input:  nil,
+			output: []byte{0x80},
+		},
+		// Not sure how to represent "the integer 0" resulting in "[]byte{0x80}"
```

Comment:
This might help?..

<img width="733" alt="Screenshot 2023-01-12 at 19 56 45" src="https://user-images.githubusercontent.com/43061995/212093039-8ecc95c8-938f-4e56-b25d-d6b957af2a4c.png">